### PR TITLE
Bugfix/alias late binding group

### LINKED ISSUE #34: Inconsistent state of `Server` and `Binding`
Hello!
I've noticed a little problem, which appears when the `Router::resolve` method is being called.

This method finds the appropriate controller or method `Binding` and resolves specific `Route`.

The problem is that `Binding` can be created before `Router` gets knowledge about middleware aliases. So `Binding` will contain different middleware aliases than `Router`.

**Doesn't work properly:**
```PHP
$server->router()->bindController(...); // creates Binding here
$server->registerMiddlewareAliases(...); // sets middleware aliases to router, Bindings won't know about them
```

**Works fine:**
```PHP
$server->registerMiddlewareAliases(...); // sets middleware aliases
$server->router()->bindController(...); // new Binding will get actual aliases
```

Currently, I didn't research a good solution. As an option, I suppose it would be good to try:
- remove `MiddlewaresCollection` from `Binding`s constructor
- pass them during calling `Router::resolve`

```PHP
// \Upgate\LaravelJsonRpc\Server\Router::resolve
public function resolve(string $method): RouteContract
{
    return $this
        ->findBinding($method)
        ->setMiddlewaresCollection($this->middlewaresCollection)
        ->resolveRoute($method);
}
```

Also, I didn't research the problem deeply, so I can't say for sure, is that a problem with an inconsistent state exists only for middleware aliases.

I appreciate any feedback you may have.

### PR DESCRIPTION
**Pull request by kbaryshnikov**
Fixes #34

### CODE REVIEW

File: src/Server/Router.php [hunk 1]
```diff
@@ -113,12 +114,16 @@ public function addMiddlewares(array $middlewares): RouteRegistryInterface
...
     /**
-     * @param MiddlewareAliasRegistryInterface|null $aliases
+     * @param array|MiddlewareAliasRegistryInterface|null $aliases
      * @return RouteRegistryInterface
      */
-    public function setMiddlewareAliases(MiddlewareAliasRegistryInterface $aliases = null): RouteRegistryInterface
+    public function setMiddlewareAliases($aliases = null): RouteRegistryInterface
     {
-        $this->middlewaresCollection->setMiddlewareAliases($aliases);
+        if (is_array($aliases)) {
```
Comment (s-chizhik) [inline src/Server/Router.php:122]: I'm not sure if method `set` should behave like `add`

**Review by s-chizhik — COMMENTED** [review]
Everything else looks good.

File: src/Server/Router.php (hunk 1 above)
Comment (kbaryshnikov) [reply src/Server/Router.php:122]: Yep... But making it behave like `add` in certain cases is what fixes the issue without a BC break.

Any better ideas?

Comment (s-chizhik) [reply src/Server/Router.php:122]: Oh, I see. I have no better idea 🤷‍♂️

Comment (kbaryshnikov) [reply src/Server/Router.php:122]: Okay, I'll merge then. It's better than nothing, anyway

**Pull request merged**